Scripts/Source/zadlibs_UDPatch.psc

```diff
@@ -826,7 +826,8 @@ Function EndThirdPersonAnimation(actor akActor, bool[] cameraState, bool permitR
 			UDCDMain.Error("EndThirdPersonAnimation("+GetActorName(akActor)+") - Actor is not loaded (Or is otherwise invalid). Aborting.")
 			return
 		EndIf
-		Armor loc_shield = StorageUtil.GetFormValue(akActor,"UD_UnequippedShield",none) as Armor
+		Form loc_shield = StorageUtil.GetFormValue(akActor,"UD_UnequippedShield",none)
+		StorageUtil.UnsetFormValue(akActor,"UD_UnequippedShield")
```

Comment:
Please, move the function inside loc_shield condition, so it doesn't unset the form if its not set.